tests/datasource/data_connector/test_configured_asset_glue_catalog_data_connector.py

```diff
@@ -762,441 +769,546 @@ def test_get_batch_data_and_metadata_with_sampling_method__random_in_batch_spec_
 
 
 def test_get_batch_data_and_metadata_with_sampling_method__mod(
+    in_memory_runtime_context,
     test_cases_for_aws_glue_data_catalog_data_connector_spark_execution_engine,
     glue_titanic_catalog,
 ):
     execution_engine = (
         test_cases_for_aws_glue_data_catalog_data_connector_spark_execution_engine
     )
-    my_data_connector = ConfiguredAssetAWSGlueDataCatalogDataConnector(
-        name="my_glue_catalog_data_connector",
-        datasource_name="FAKE_Datasource_NAME",
-        execution_engine=execution_engine,
-        assets={
-            "db_test.tb_titanic": {
-                "table_name": "tb_titanic_with_partitions",
-                "database_name": "db_test",
-            }
+    in_memory_runtime_context.datasources["FAKE_Datasource_NAME"] = Datasource(
```

Comment:
Oh, I hadn't opened up the large diff file. This looks repeated, could you make a util method or fixture for this and the `BatchDefinition` below?